qgsquick/from_qgis/quickgui/plugin/qgsquickfeatureform.qml

```diff
@@ -199,38 +173,33 @@ Item {
   }
 
   function save() {
-    if ( !model.constraintsHardValid )
+    if ( !controller.constraintsHardValid )
     {
       console.log( qsTr( 'Constraints not valid') )
       return
     }
-    else if ( !model.constraintsSoftValid )
+    else if ( !controller.constraintsSoftValid )
     {
       console.log( qsTr( 'Note: soft constraints were not met') )
     }
 
     parent.focus = true
     if ( form.state === "Add" ) {
-      model.create()
+      controller.create()
       state = "Edit"
     }
     else
     {
-      model.save()
+      controller.save()
     }
 
     saved()
   }
 
   function hasAnyChanges() {
-    return form.model.attributeModel.hasAnyChanges()
+    return form.controller.hasAnyChanges()
```

Comment:
`return form.controller.hasAnyChanges` since its Q_property now,  not a function.